types/ramda/index.d.ts

```diff
@@ -777,6 +777,7 @@ declare namespace R {
          * matches, fn returns undefined.
          */
         cond(fns: ReadonlyArray<[Pred, (...a: any[]) => any]>): (...a: any[]) => any;
+        cond<A, B>(fns: ReadonlyArray<[SafePred<A>, (...a: A[]) => B]>): (...a: A[]) => B;
```

Comment:
Unless I'm mistaken, this overload will only be used if explicit type arguments are provided.  Was that your expectation?

Reply:
It was, unless there's another method I'm not aware of there's no way for TS to reliably infer the type here without breaking existing code. Else `tsc` infers A & B as `never`.

Opting in to the type allows people to type the cond if they want or preserve their existing assumptions about types.

Reply:
It seems reasonable - I just didn't want it to be a nasty surprise. :wink: